test/e2e/tas/rayjob_test.go

```diff
@@ -0,0 +1,264 @@
+									},
+									Limits: corev1.ResourceList{
+										corev1.ResourceCPU: resource.MustParse("50m"),
+									},
+								},
+							},
+						},
+						RestartPolicy: corev1.RestartPolicyNever,
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							kueuealpha.PodSetRequiredTopologyAnnotation: testing.DefaultBlockTopologyLevel,
+						},
+					},
+				}).
+				Image(rayv1.HeadNode, kuberayTestImage, []string{}).
+				Image(rayv1.WorkerNode, kuberayTestImage, []string{}).
+				Obj()
+
+			gomega.Expect(k8sClient.Create(ctx, rayjob)).Should(gomega.Succeed())
+
+			ginkgo.By("RayJob is unsuspended", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(rayjob), rayjob)).To(gomega.Succeed())
+					g.Expect(rayjob.Spec.Suspend).Should(gomega.BeFalse())
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			pods := &corev1.PodList{}
+			ginkgo.By("ensure all pods are created", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(rayjob.Namespace))).To(gomega.Succeed())
+					g.Expect(pods.Items).Should(gomega.HaveLen(numPods))
+					// timeout is super long to ensure all cluster pods are up and running - rayimage takes long time to load and then to sync with head
+				}, util.StartUpTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("ensure all pods are scheduled", func() {
+				listOpts := &client.ListOptions{
+					FieldSelector: fields.OneTermNotEqualSelector("spec.nodeName", ""),
+				}
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name), listOpts)).To(gomega.Succeed())
+					g.Expect(pods.Items).Should(gomega.HaveLen(numPods))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("verify the assignment of pods are as expected with rank-based ordering", func() {
+				gomega.Expect(k8sClient.List(ctx, pods, client.InNamespace(ns.Name))).To(gomega.Succeed())
+				gotWorkersAssignment := readRankAssignmentsFromRayJobWorkerPods(pods.Items)
+				// raycluster worker pods does not contain the index, thus what we can check is
+				// how many raycluster worker pods are assigned to each node
+				wantAssignment := map[string]int{
+					"kind-worker":  1,
+					"kind-worker2": 1,
+					"kind-worker3": 1,
+				}
+				gomega.Expect(wantAssignment).Should(gomega.BeComparableTo(gotWorkersAssignment))
+			})
+
+			ginkgo.By("verify the submitter pod topology annotation", func() {
+				domain, level := readTopologyAnnotationFromRayJobSubmitterPod(pods.Items)
+				gomega.Expect(domain).Should(gomega.Equal(kueuealpha.PodSetRequiredTopologyAnnotation))
+				gomega.Expect(level).Should(gomega.Equal(testing.DefaultBlockTopologyLevel))
+			})
+		})
+	})
+})
+
+func readRankAssignmentsFromRayJobWorkerPods(pods []corev1.Pod) map[string]int {
```

Comment:
This name is very confusing. Let's just return the set of used nodes. Maybe "readAssignedNodes", which just returns the set of ".spec.NodeName"'s used by the worker pods.